services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -311,63 +311,51 @@ async def add_project(
                 prj["tags"] = []
             return prj
 
-    async def load_user_projects(
+    async def load_projects(
         self,
-        user_id: int,
-        filter_by_services: Optional[List[Dict]] = None,
-    ) -> List[Dict]:
-        log.info("Loading projects for user %s", user_id)
-        async with self.engine.acquire() as conn:
-            user_groups: List[RowProxy] = await self.__load_user_groups(conn, user_id)
-            query = textwrap.dedent(
-                f"""\
-                SELECT *
-                FROM projects
-                WHERE projects.type != 'TEMPLATE'
-                AND (jsonb_exists_any(projects.access_rights, array[{', '.join(f"'{group.gid}'" for group in user_groups)}])
-                OR prj_owner = {user_id})
-                """
-            )
-            projects_list = await self.__load_projects(
-                conn, query, user_id, user_groups, filter_by_services=filter_by_services
-            )
-
-        return projects_list
-
-    async def load_template_projects(
-        self,
-        user_id: int,
+        user_id: PositiveInt,
         *,
-        only_published=False,
+        filter_by_project_type: Optional[ProjectType] = None,
         filter_by_services: Optional[List[Dict]] = None,
-    ) -> List[Dict]:
-        log.info("Loading public template projects")
-
-        # TODO: eliminate this and use mock to replace get_user_project instead
-        projects_list = [prj.data for prj in Fake.projects.values() if prj.template]
-
+        only_published: Optional[bool] = False,
+        offset: Optional[int] = 0,
+        limit: Optional[int] = None,
+    ) -> Tuple[List[Dict[str, Any]], List[ProjectType], int]:
+        project_type_where_clause = (
+            f"= '{filter_by_project_type.value}'"
+            if filter_by_project_type
+            else "IS NOT NULL"
+        )
         async with self.engine.acquire() as conn:
             user_groups: List[RowProxy] = await self.__load_user_groups(conn, user_id)
-
-            # NOTE: in order to use specific postgresql function jsonb_exists_any we use raw call here
             query = textwrap.dedent(
```

Comment:
I think there is not need of the dedent and can be multiline string as
```python
"SELECT *"
f"WHERE project.type {project_type_where_clause}"
...
```

Don't you need ``sqlalchemy.sql.text``? SEE [this](https://github.com/ITISFoundation/osparc-simcore/blob/5d507ee62337d209153120cfd913e1e1039414a6/services/storage/src/simcore_service_storage/access_layer.py#L169)

Reply:
I understand that some functions like ``jsonb_exists_any`` is not available in sqlalchemy, but having all text, should not be insecure? Is there a way to mix both sqlalchemy expressions and text? 

Something like `` project.select().where( _and( projects.c.type == ..., text() ) ...``
See e.g. [here](https://github.com/ITISFoundation/osparc-simcore/blob/c693b9fc74e66525398bab633f1f8d6a39fda677/services/web/server/src/simcore_service_webserver/studies_dispatcher/_core.py#L67) how you can compose the statement  ``stmt`` depending on different conditions by aggregating "pieces"

Reply:
the insecure comment I'm not sure I understand why? is sqlalchemy not doing exactly this for us? in the end it sends full SQL commands.
I'll check it out.

Reply:
The basic idea is that with string replacement somebody could basically inject whatever and change completely the statement i.e.  ``user_id = 'SELECT password ... '''